prism/src/main/java/org/apache/falcon/resource/AbstractInstanceManager.java

```diff
@@ -485,8 +485,17 @@ public int compare(Instance i1, Instance i2) {
                             : end2.compareTo(end1);
                 }
             });
-        }//Default : no sort
-
+        }
```

Comment:
I think "else" is missing here. Look like last block of sorting should be under "else".  

Also since the code is redundant. You can simple change the condition on line 468 to if (orderBy.equals("starttime") || orderBy.isEmpty())